microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshots.md

```diff
@@ -119,7 +119,7 @@ For each constructor in the list, the **Summary** view also shows a number such
 | Column name | Description |
 |:--- |:--- |
 | **Distance** | Displays the distance to the root using the shortest simple path of nodes.  See [Distance](./memory-101.md#distance) in _Memory terminology_. |
-| **Shallow size** | Displays the sum of shallow sizes of all objects created by a certain constructor function.  The shallow size is the size of memory held by an object (generally, arrays and strings have larger shallow sizes).  See [Shallow size](./memory-101.md#shallow-size) in _Memory terminology_. |
+| **Shallow size** | Displays the sum of shallow sizes of all objects created by a certain constructor function.  The shallow size is the size of memory directly held by an object. Most objects have a small shallow size because they only store their own object descriptions in memory, while their values are stored elsewhere in the renderer memory. See [Shallow size](./memory-101.md#shallow-size) in _Memory terminology_. |
```

Comment:
```suggestion
| **Shallow size** | Displays the sum of shallow sizes of all objects created by a certain constructor function.  The shallow size is the size of memory directly held by an object. Most objects have a small shallow size, because they only store their own object descriptions in memory, while their values are stored elsewhere in the renderer memory. See [Shallow size](./memory-101.md#shallow-size) in _Memory terminology_. |
```
if no comma, the sentence means that the reason most objects have small shallow size is because they only store...  it's giving the reason why most objects have a small shallow size, as opposed to why other objects have a small shallow size.
if comma, the sentence means that most objects have small shallow size, and that is because they only store...

Reply:
probably need comma: change
Most JavaScript objects store their values elsewhere in the renderer memory
to
Most JavaScript objects store their values elsewhere, in the renderer memory

@captainbrosset , are the descriptions stored in the renderer memory, or in the heap?  We shouldn't say just "memory", given that we're talking about different areas or types of memory.  Need always a qualifier: heap memory vs renderer memory.  if i understand.

Reply:
keep the word "own"?  what's the opposite: description of a contained object, vs. description of the present (own) object?

Reply:
The "one place" and the "other place" are both within the JS heap in the renderer memory.

Reply:
The JS heap is _a region of_ the renderer memory. The renderer memory is the memory that's used by a rendered page, and which contains the JS heap, as well as other things.
When we say that most objects store their values in a different place in the renderer memory, I think that place is outside of the JS heap.

Reply:
Also, I don't think that we absolutely need to say "the JavaScript heap in the renderer memory". I don't know that developers really need to know where the JS heap is. It's fine that the memory-101 article says this in the **JavaScript heap and renderer memory** section, but I don't know that repeating it later really helps understanding.